### DISCUSSION TITLE
Bug Report: Changing scaling mode of selected transformation of vector layer deselects

### DISCUSSION THREAD

**Discussion by miya-sans** [keiwando/toonsquid#310] [Q&A] [answered]
Device: iPad Pro (12.9 inch) (6th generation)
iOS Version: 16.4.1
Version: 1.8.6

https://github.com/keiwando/toonsquid/assets/42623210/de277671-0137-4734-a03d-05c1ea42853f

For raster layers, changing the scaling mode of a selected transformation will merely change the selected mode, allowing you to continue transforming. However, for vector layers, following the same steps, changing the scaling mode of a selected transformation, will deselect your transformation.

In addition, the scaling mode will default back to uniform when re-opening the project, this is disadvantageous, as uniform is the least used scaling method for certain workflows and having to re-select the scaling method is redundant. 

Also, on a raster layer if you select perspective or warp as the scaling method, and then select a vector layer, the scaling method for both raster and vector transformations will default back to uniform. Of course, this is because perspective and warp transform is not in vector layers. Having vector and raster layers remember its scaling mode would be a solution. This resolves the difficulty of using both raster and vector layers while using the warp and perspective scaling modes.

**Comment by keiwando** [comment, marked as answer]
The accidental deselection should now be fixed with the 1.9 update. Thanks for the detailed report and for the feedback on the scaling mode setting for vector and pixel editing. I've noted it down.
